gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageGrpcIntegrationTest.java

```diff
@@ -0,0 +1,206 @@
+package com.google.cloud.hadoop.gcsio.integration;
+
+import static com.google.cloud.hadoop.gcsio.integration.GoogleCloudStorageTestHelper.assertObjectContent;
+import static com.google.cloud.hadoop.gcsio.integration.GoogleCloudStorageTestHelper.writeObject;
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.Assert.assertThrows;
+
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorage;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageImpl;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageItemInfo;
+import com.google.cloud.hadoop.gcsio.ListObjectOptions;
+import com.google.cloud.hadoop.gcsio.StorageResourceId;
+import com.google.common.collect.Lists;
+import com.google.common.flogger.GoogleLogger;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.SeekableByteChannel;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class GoogleCloudStorageGrpcIntegrationTest {
+
+  // Prefix this name with the prefix used in other gcs io integrate tests once it's whitelisted by
+  // GCS to access gRPC API.
+  private static final String BUCKET_NAME = "gcs-grpc-team-dataproc-it";
+
+  private static GoogleCloudStorage createGoogleCloudStorage() {
+    try {
+      return new GoogleCloudStorageImpl(
+          GoogleCloudStorageTestHelper.getStandardOptionBuilder().setGrpcEnabled(true).build(),
+          GoogleCloudStorageTestHelper.getCredential());
+    } catch (IOException e) {
+      GoogleLogger logger = GoogleLogger.forEnclosingClass();
+      logger.atWarning().withCause(e).log(
+          "Caught exception during GCS (%s) buckets creation", BUCKET_NAME);
+      return null;
+    }
+  }
+
+  @BeforeClass
+  public static void createBuckets() throws IOException {
+    GoogleCloudStorage rawStorage = createGoogleCloudStorage();
+    rawStorage.createBucket(BUCKET_NAME);
+  }
+
+  @AfterClass
+  public static void cleanupBuckets() throws IOException {
+    GoogleCloudStorage rawStorage = createGoogleCloudStorage();
+    // randomize buckets order in case concurrent clean ups are running
+    List<GoogleCloudStorageItemInfo> objectsToDelete =
+        rawStorage.listObjectInfo(
+            BUCKET_NAME, /* objectNamePrefix= */ null, ListObjectOptions.DEFAULT_FLAT_LIST);
+
+    rawStorage.deleteObjects(
+        Lists.transform(objectsToDelete, GoogleCloudStorageItemInfo::getResourceId));
+    rawStorage.deleteBuckets(Arrays.asList(BUCKET_NAME));
```

Comment:
Cleanup in integration test helper is more complicated because it deletes leaked buckets as well, not only current test bucket, ideally we should reuse it here.

Reply:
Done.